bin/ci/test_helpers.sh

```diff
@@ -0,0 +1,64 @@
+# This file is a collection of commonly used functions in testing core-plans
+
+# Attempts to start a supervisor.
+# Detects if a supervisor is already running and emits a warning if one is found
+# Shares logging path with the studio supervisor.
+# This is to allow local testing of packages, which will likely happen in a
+# studio, to behave as close as possible to CI.
+ci_ensure_supervisor_running() {
+  # We can't use `hab sup status` as the exit code for `not running` vs.
+  # `running but recieved an error` is the same. In the case of the later
+  # we'll catch the error in the retry logic that ensures the supervisor
+  # is running before handing control back to the tests
+  #
+  # The studio doesn't contain ps by default so we need to use the presence
+  # of the LOCK file to attempt to determine if the supervisor is running.
+  #
+  # Failure case: Supervisor is stopped, but LOCK file is still present
+  if [ -f /hab/sup/default/LOCK ]; then
+    echo "--- :warning: Supervisor LOCK file found, not starting the supervisor"
+  else
+    echo "--- :habicat: Starting the supervisor"
+    if [ ! -f /hab/sup/default/sup.log ]; then
+      mkdir -p /hab/sup/default
+      touch /hab/sup/default/sup.log
+    fi
+    hab sup run > /hab/sup/default/sup.log 2>&1 &
+  fi
+
+  for retry in {1..20}; do
+    sleep 1
+    hab sup status > /dev/null 2>&1 && break
+    echo "Waiting for Supervisor to start: $retry"
+  done
+
+  if [ "$retry" -eq 5 ]; then
+    echo "--- :boomboom: Unable to connect to supervisor"
+    exit 1
+  fi
+}
+
+# Loads the specified service
+# Retries until the service shows running or it specified timeout (default 5 seconds)
+# period has passed.
+ci_load_service() {
```

Comment:
This currently won't work for any service loading that requires or benefits from binds (haproxy comes to mind).

Maybe this could be reduced to ci_service_loaded(ident, wait_time), allowing the individual test to do the service loading, and the common script to do the check/fail.

Reply:
I think those sorts of services should be the exception, and potentially highlights a complexity problem. If a core package requires more than `hab svc load` + a wait for it to start,  I would probably argue that the Plan is doing too much, or perhaps it doesn't belong in core. There may be exceptions to this, but I don't think we've run into any yet. 

Digressing a bit, I personally believe that binds don't belong in core-plans, that feels like something that belongs in a wrapper. Optional binds would be fine for things like haproxy as it would be useful for dev scenarios, but required binds feels more like a deployment decision which core should be relatively un-opinionated on.

Reply:
👍